### DISCUSSION TITLE
Trouble for onvert Mat to ImageSource in Maui .Net8.0 android

### DISCUSSION THREAD

**Discussion by sergiofreitasbra** [emgucv/emgucv#969] [Q&A]
Hi everyone im trying for days convert Mat to ImageSource in Maui .Net8.0 android, with out any sucess Somebody can help how convert and binding to a image source from viewmodel thank you very much

**Comment by emgucv** [comment]
If you are trying to display a Mat in MUAI, the simplest way is to render the Mat under a CvImageView:
https://github.com/emgucv/emgucv/blob/ecce3ab64e089277db1521836472e27fba844c58/Emgu.CV.Example/MAUI/BuildInfo/MainPage.xaml#L47
You will need to import the UI from the Emgu CV MAUI runtime nuget package like this:
https://github.com/emgucv/emgucv/blob/ecce3ab64e089277db1521836472e27fba844c58/Emgu.CV.Example/MAUI/BuildInfo/MainPage.xaml#L4
This will be compatible with all MAUI platforms.

The implementation of CvImageView can be found here:
[Emgu.CV.Runtime/Maui/UI/CvImageView.cs](https://github.com/emgucv/emgucv/blob/ecce3ab64e089277db1521836472e27fba844c58/Emgu.CV.Runtime/Maui/UI/CvImageView.cs#L1)

**Reply by sergiofreitasbra** [reply]
Hi thank you very much finally i did get working, only problem the image is all gray and the wrong size do you know what could cause this thank you again

**Reply by emgucv** [reply]
Do you have a small project that can re-produce the issue? We can look into that if we can reproduce it on our side.

**Reply by sergiofreitasbra** [reply]
Hello yes

VideoCapture vc;

vc = new VideoCapture(0, VideoCapture.API.Android);
vc.Set(CapProp.FrameWidth, 640);
vc.Set(CapProp.FrameHeight, 480);
Mat frame = new Mat();
try
{
    
        if (vc != null && vc.Read(frame))
        {                                    
            var frame1 = new Image<Bgr, byte>(frame);
            Imageview.SetImage(frame);
        }
}
catch (Exception ex)
{


}

thank you

**Reply by emgucv** [reply]
Open CV's VideoCapture implementation for Android is experimental and quite buggy. It has many issues that will cause access violation and crashing issues. 

VideoCapture class also is not implemented on iOS with Open CV.

If you are using MAUI, you can use platform specific native camera capture API to capture the video frames.  e.g.
https://github.com/emgucv/emgucv/blob/master/Emgu.CV.Runtime/Maui/UI/ProcessAndRenderPage.cs

The code is included in Emgu.CV.runtime.maui package.

**Reply by sergiofreitasbra** [reply]
Hello thank you very much i will check it :)
